File: 1-10/3/3.cpp
#include <iostream>
bool isprime(unsigned long long v);

using namespace std;
int main()
{
    unsigned long long num = 600851475143,max = 0, x;
    
    for(x = 2 ; (x * x) <= num ; x++)
    {    if((num % x) == 0)
            if(isprime(x))
                if(x > max)
                    max = x;         
    }
    cout<<"Answer = "<<max<<endl;
    return 0;
}

bool isprime(unsigned long long v)
{
    unsigned long long i = 0;
    bool rv = true;
    
    if(v != 1)
        for(i = 2 ; i <= (v / 2) ; i++)
        {
            if(v % i == 0)
                 rv = false;  
        }   
    else
        rv = false;
   
    return rv;
}

File: 1-10/3/3.py
def func():
	num = 600851475143
	i = 3
	itr=1
	largestPrimeFactor = 1
	while(i < num/2):
		print("iteration-",itr)
		if(i % num == 0 and isPrime(i) and i > largestPrimeFactor):
			largestPrimeFactor = i
		itr+=1	
	print largestPrimeFactor		


def isPrime(num):
	c = 3
	while(c < num/2):
		if(c%num == 0):
			return false
		c+=2			
	return true		

if __name__ == '__main__':
	func()